router/src/validation.rs

```diff
@@ -352,36 +368,31 @@ fn tokenizer_worker(tokenizer: Tokenizer, receiver: flume::Receiver<TokenizerReq
 
 /// Get input length and optionally truncate it
 fn prepare_input(
-    inputs: String,
+    mut inputs: String,
     truncate: Option<usize>,
     tokenizer: &Tokenizer,
-) -> Result<(String, usize), ValidationError> {
+) -> Result<(tokenizers::Encoding, String), ValidationError> {
     // Get the number of tokens in the input
     let mut encoding = tokenizer
-        .encode(inputs.clone(), true)
+        .encode(inputs.clone(), false)
```

Comment:
The documentation for `add_special_tokens` is pretty unclear, but is there a difference in the generated input lengths when this param is set to `false`. The one thing I would want to double-check here is that we don't under-count the number of tokens in the input during validation, otherwise we could exceed the max positional embeddings during inference and cause segfaults.

Reply:
That's right.
Due to a simple test during development, the default value of `add_special_tokens` was changed and I forgot to revert it. I'll fix it back to the original.